File: AdvancedSkill/src/main/java/com/example/advancedskill/test/SecondActivity.java
package com.example.advancedskill.test;

import androidx.appcompat.app.AppCompatActivity;

import android.app.ActivityManager;
import android.graphics.BitmapFactory;
import android.os.Bundle;
import android.widget.TextView;

import com.example.advancedskill.activity.IntentParcelableActivity;
import com.example.advancedskill.activity.IntentSerializableActivity;
import com.example.advancedskill.entity.Person;
import com.example.advancedskill.entity.User;
import com.example.advancedskill.R;

public class SecondActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_second);

        TextView tv_text = findViewById(R.id.tv_text);
        String way = getIntent().getStringExtra("way");

        if (way != null){
            if (way.equals(IntentSerializableActivity.way)){
                //调用了Intent的getSerializableExtra()方法来获取通过参数传递过来的序列化对象，接着再将它向下转型成Person对象
                Person person = (Person) getIntent().getSerializableExtra("person_data");
                tv_text.setText(way +"方式："+ person.toString());
            }else if (way.equals(IntentParcelableActivity.way)){
                //调用getParcelableExtra()方法来获取传递过来的对象，其他的地方完全相同
                User user = (User)getIntent().getParcelableExtra("person_data");
                tv_text.setText(way +"方式："+ user.toString());
            }
        } else {
            tv_text.setText("第二行代码");
        }

        setCustomTaskDescription();
    }

    private void setCustomTaskDescription() {
        ActivityManager.TaskDescription taskDescription = new ActivityManager.TaskDescription("SecondActivity",
                BitmapFactory.decodeResource(getResources(), R.drawable.zhiboba));
        setTaskDescription(taskDescription);
    }
}

File: Activity/src/main/java/com/example/activity/SecondActivity.kt
package com.example.activity

import android.content.Context
import android.content.Intent
import android.os.Bundle

class SecondActivity : BaseActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_second)
    }

    //所有定义在companion object中的方法都可以使用类似于Java 静态方法的形式调用
    companion object {
        fun actionStart(context: Context,data1: String,data2: String){
            val intent = Intent(context,SecondActivity::class.java)
            intent.putExtra("param1",data1)
            intent.putExtra("param2", data2)
            context.startActivity(intent)
        }
    }
}